src/Kiota.Builder/Writers/Ruby/CodeClassDeclarationWriter.cs

```diff
@@ -24,12 +23,17 @@ public override void WriteCodeElement(CodeClass.Declaration codeElement, Languag
                 writer.WriteLine($"require_relative '{codeUsing.Key.ToSnakeCase()}'");
             writer.WriteLine();
             if(codeElement?.Parent?.Parent is CodeNamespace) {
-                writer.WriteLine($"module {NormalizeNameSpaceName(codeElement.Parent.Parent.Name)}");
+                writer.WriteLine($"module {codeElement.Parent.Parent.Name.NormalizeNameSpaceName("::")}");
                 writer.IncreaseIndent();
             }
-            if ("entity".Equals(codeElement?.Inherits?.Name, StringComparison.OrdinalIgnoreCase)){
-                codeElement.Inherits.Name = "Graphrubyv4::Utilities::Users::Entity";
+            
+            if("entity".Equals(codeElement?.Inherits?.Name.ToFirstCharacterLowerCase())) {
+                int index = codeElement.Parent.Parent.Name.ToFirstCharacterLowerCase().LastIndexOf("users")+7;
```

Comment:
I'm not a big fan of magic numbers. Maybe store entity as a constant so you can:
- use it for the comparison in the if
- use the length here
- use it below when building the new reference

Also, I'd have to double check, but I think this patching was only required before I fixed the back references in the builder. It might not be needed anymore. (I'd have to remove it in the other languages as well)